new/chef-upgrade.md

```diff
@@ -0,0 +1,70 @@
+---
+RFC: unassigned
+Author: Thom May <thom@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Chef should be able to upgrade itself
+
+Users should be able to automatically upgrade to the desired version of chef without having to use third party cookbooks. The upgrade should occur without affecting the normal chef client run, ensuring that a consistent state is preserved.
+
+## Motivation
+
+```
+  As an operator, I want chef versions to be consistent across my infrastructure,
+  so that I can ensure my code works correctly
+```
+```
+  As an operator, I want to easily and safely upgrade chef to my desired version,
+  so that I can easily roll out new versions.
+```
+```
+  As a developer, I want our users to get new features and bug fixes in a timely and safe manner,
+  so that everyone's happy.
+```
+
+## Specification
+
+At the end of each chef run, just before the reboot handler, the chef client should check with an update service to discern whether there is a new version of chef. If there is, the client should decide whether or not to upgrade, based on configuration supplied by the administrator.
```

Comment:
@coderanger I originally wrote the RFC so that the upgrade came first - before any part of the chef run. But I've come round to the idea that it's worth doing after, so that the user can provide configuration to the upgrade as part of the chef run. And now that we can run stuff as a handler, we can do it after a complete chef run.
I'm not convinced the timing will be a huge problem; it's likely that the chef upgrade will happen some time before any new feature is used.